clear the random stack value

Data type SaHpiAnnouncementT has holes and pads between its elements, and both
announcement and getAnnouncement in saHpiAnnunciatorGet/3.c come from stack,
and include the random value, these random value will not be set by 
saHpiAnnunciatorGet, and lead to memcmp return false.

Samilar root cause in saHpiAnnunciatorAdd/3.c, the random value in announcement
will lead to parameter check failure in addAnnouncement

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
---
 HPI-B.01.01/src/annunciator/saHpiAnnunciatorAdd/3.c |    1 +
 HPI-B.01.01/src/annunciator/saHpiAnnunciatorGet/3.c |    2 ++
 2 files changed, 3 insertions(+)

diff --git a/HPI-B.01.01/src/annunciator/saHpiAnnunciatorAdd/3.c b/HPI-B.01.01/src/annunciator/saHpiAnnunciatorAdd/3.c
index d757fd5..0f7709f 100644
--- a/HPI-B.01.01/src/annunciator/saHpiAnnunciatorAdd/3.c
+++ b/HPI-B.01.01/src/annunciator/saHpiAnnunciatorAdd/3.c
@@ -50,6 +50,7 @@ int processAnnunRdr(SaHpiSessionIdT sessionId,
 	SaHpiSeverityT *severity;
 	int severityCount;
 
+	memset(&announcement, 0, sizeof(announcement));
 	status = setWriteMode(sessionId, resourceId, annunRec, &mode);
 	if (status == ANNUN_ERROR) {
 		e_trace();
diff --git a/HPI-B.01.01/src/annunciator/saHpiAnnunciatorGet/3.c b/HPI-B.01.01/src/annunciator/saHpiAnnunciatorGet/3.c
index 7e25718..4cc4202 100644
--- a/HPI-B.01.01/src/annunciator/saHpiAnnunciatorGet/3.c
+++ b/HPI-B.01.01/src/annunciator/saHpiAnnunciatorGet/3.c
@@ -48,6 +48,8 @@ int run_test1(SaHpiSessionIdT sessionId,
 	int retval = SAF_TEST_NOTSUPPORT;
 	int count = 0;
 
+	memset(&announcement, 0, sizeof(announcement));
+	memset(&getAnnouncement, 0, sizeof(getAnnouncement));
 	announcement.EntryId = SAHPI_FIRST_ENTRY;
 
 	while (status != SA_ERR_HPI_NOT_PRESENT
-- 
1.7.10.4

